### PR TITLE
topology: Add CML topology support for chromebook rt5682+max98357a va…

### PR DESCRIPTION
**Pull request by sathyap-chrome**
…riant

Add new PCMs and backends for CML chromebook with codecs rt5682+max98357a.

Signed-off-by: Sathya Prakash M R <sathya.prakash.m.r@intel.com>

### CODE REVIEW

File: tools/topology/sof-cml-rt5682_max98357a.m4 [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	
+# Low Latency playback pipeline 2 on PCM 1 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-src-playback.m4,
+	3, 1, 2, s32le,
+	48, 1000, 0, 0)
+
+# Passthrough capture pipeline 4 on PCM 2 using max 4 channels.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+#PIPELINE_PCM_ADD(sof/pipe-volume-capture.m4,
```
Comment (ranj063) [inline tools/topology/sof-cml-rt5682_max98357a.m4]: @sathyap-chrome lets remove this commented line

File: tools/topology/sof-cml-rt5682_max98357a.m4 [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+PIPELINE_PCM_ADD(sof/pipe-passthrough-capture.m4,
+	4, 2, 4, s32le,
+	48, 1000, 0, 0)
+
+# Low Latency playback pipeline 5 on PCM 3 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+# PIPELINE_PCM_ADD(sof/pipe-passthrough-playback.m4,
```
Comment (ranj063) [inline tools/topology/sof-cml-rt5682_max98357a.m4]: same here

File: tools/topology/sof-cml-rt5682_max98357a.m4 [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+        5, 3, 2, s32le,
+        48, 1000, 0, 0)
+
+# Low Latency playback pipeline 6 on PCM 4 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+# PIPELINE_PCM_ADD(sof/pipe-passthrough-playback.m4,
```
Comment (ranj063) [inline tools/topology/sof-cml-rt5682_max98357a.m4]: same here

File: tools/topology/sof-cml-rt5682_max98357a.m4 [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	6, 4, 2, s32le,
+	48, 1000, 0, 0)
+
+# Low Latency playback pipeline 7 on PCM 5 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+# PIPELINE_PCM_ADD(sof/pipe-passthrough-playback.m4,
```
Comment (ranj063) [inline tools/topology/sof-cml-rt5682_max98357a.m4]: and here

**Comment by ranj063** [comment]
@sathyap-chrome LGTM. only minor comments about removing stale commented lines

**Comment by lgirdwood** [comment]
@sathyap-chrome please also fix conflicts, we no longer use autotools for build.

File: tools/topology/sof-cml-rt5682_max98357a.m4 (hunk 3 above)
Comment (sathyap-chrome) [inline tools/topology/sof-cml-rt5682_max98357a.m4]: Addressed all comments and modified the Speaker DAI setting to an LRCLK of 46875.
The new PR is here : https://github.com/thesofproject/sof/pull/1138

Comment (lgirdwood) [reply tools/topology/sof-cml-rt5682_max98357a.m4]: Fwiw, best to just force push your new changes rather than creating a new PR. I've closed this one.

**Pull request closed without merging**